src/se/vidstige/jadb/managers/KeyCode.java

```diff
@@ -0,0 +1,1328 @@
+package se.vidstige.jadb.managers;
+
+/**
+ * Code used from Android KeyEvent class.
+ *
+ * Copyright (C) 2006 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Author: pablobaxter.
+ */
```

Comment:
Alright, great. Can you also provide a link with the original source at the top of the file to make it easier to update, etc.?